src/log.ts

```diff
@@ -176,8 +176,12 @@ export namespace message {
       channel === 'x' ? 'width' : 'height'} is provided.`;
   }
 
-  export function scaleTypeNotWorkWithChannel(channel: Channel, scaleType: ScaleType, newScaleType: ScaleType) {
-    return `Channel ${channel} does not work with ${scaleType} scale. We are using ${newScaleType} scale instead.`;
+  export function cannotOverrideBinScaleType(channel: Channel) {
+    return `Cannot override scale type for binned channel ${channel}.`;
+  }
+
+  export function scaleTypeNotWorkWithChannel(channel: Channel, scaleType: ScaleType) {
```

Comment:
Because we had to compute the right scale twice. You can just look at the vega output.